mesonpy/__init__.py

```diff
@@ -949,45 +994,18 @@ def _meson_version(self) -> str:
 
     @property
     def name(self) -> str:
-        """Project name. Specified in pyproject.toml."""
-        name = self._metadata.name if self._metadata else self._meson_name
-        assert isinstance(name, str)
-        return name.replace('-', '_')
+        """Project name."""
+        return str(self._metadata.name).replace('-', '_')
```

Comment:
`str` is not needed here. The assert was because mypy was complaining because of the import IIRC.

Reply:
mypy in CI disagrees with you https://github.com/mesonbuild/meson-python/actions/runs/4359109213/jobs/7620540144#step:5:13

Reply:
That field can only ever be a string.

https://pep621.readthedocs.io/en/latest/#pyproject_metadata.StandardMetadata.name

I don't think mypy is loading the type hint correctly, hence the assert.

Reply:
It's the version of mypy used in the CI, later versions of mypy do not complain